03_Modules/Transactions/src/module/module.ts

```diff
@@ -337,11 +372,33 @@ export class TransactionsModule extends AbstractModule {
   ): Promise<void> {
     this._logger.debug('MeterValues received:', message, props);
 
-    // TODO: Add meterValues to transactions
     // TODO: Meter values can be triggered. Ideally, it should be sent to the callbackUrl from the message api that sent the trigger message
     // TODO: If sendCostUpdatedOnMeterValue is true, meterValues handler triggers cost update
     //  when it is added into a transaction
 
+    const meterValues = message.payload.meterValue;
+    const stationId = message.context.stationId;
+
+    await Promise.all(
+      meterValues.map((meterValue) =>
+        this.transactionEventRepository.createMeterValue(meterValue),
```

Comment:
Does it make sense to store the meter values if we might potentially reject the request a few lines below?

Reply:
to prevent data loss we're always storing the meter value, and in theory there shouldn't consequences to storing the data even if the signature's off... hopefully 🤔